scraper.py

```diff
@@ -0,0 +1,246 @@
+        if not isinstance(body, list):
+            raise ClientException("expecting list")
+        return [t["id"] for t in body]
+
+    async def get_timeline_ids(self, tenant_id):
+        resp = await self.sess.get(f"{self.endpoint}/v1/tenant/{tenant_id}/timeline")
+        body = await resp.json()
+        if not resp.ok:
+            raise ClientException(f"{resp}")
+        if not isinstance(body, list):
+            raise ClientException("expecting list")
+        return [t["timeline_id"] for t in body]
+
+    async def get_layer_map(self, tenant_id, timeline_id, reset) -> Tuple[Optional[str], Any]:
+        resp = await self.sess.get(
+            f"{self.endpoint}/v1/tenant/{tenant_id}/timeline/{timeline_id}/layer",
+            params={"reset": reset},
+        )
+        if not resp.ok:
+            raise ClientException(f"{resp}")
+        launch_ts = resp.headers.get("PAGESERVER_LAUNCH_TIMESTAMP", None)
+        body = await resp.json()
+        return (launch_ts, body)
+
+
+async def scrape_timeline(ps_id: str, ps_client: Client, db: asyncpg.Pool, tenant_id, timeline_id):
+    now = datetime.datetime.utcnow()
+    launch_ts, layer_map = await ps_client.get_layer_map(
+        tenant_id,
+        timeline_id,
+        # Reset the stats on every access to get max resolution on the task kind bitmap.
+        # Also, under the "every scrape does a full reset" model, it's not as urgent to
+        # detect pageserver restarts in post-processing, because, to answer the question
+        # "How often has the layer been accessed since its existence, across ps restarts?"
+        # we can simply sum up all scrape points that we have for this layer.
+        reset="AllStats",
+    )
+    await db.execute(
+        """
+            insert into layer_map (scrape_ts, pageserver_id, launch_id, tenant_id, timeline_id, layer_map)
+            values ($1, $2, $3, $4, $5, $6::jsonb);""",
+        now,
+        ps_id,
+        launch_ts,
+        tenant_id,
+        timeline_id,
+        json.dumps(layer_map),
+    )
+
+
+async def timeline_task(
+    args, ps_id, tenant_id, timeline_id, client: Client, db: asyncpg.Pool, stop_var: asyncio.Event
+):
+    """
+    Task loop that is responsible for scraping one timeline
+    """
+
+    while not stop_var.is_set():
+        try:
+            logging.info(f"begin scraping timeline {tenant_id}/{timeline_id}")
+            await scrape_timeline(ps_id, client, db, tenant_id, timeline_id)
+            logging.info(f"finished scraping timeline {tenant_id}/{timeline_id}")
+        except Exception:
+            logging.exception(f"{tenant_id}/{timeline_id} failed, stopping scraping")
+            return
+        # TODO: use ticker-like construct instead of sleep()
+        # TODO: bail out early if stop_var is set. That needs a select()-like statement for Python. Is there any?
+        await asyncio.sleep(args.interval)
+
+
+async def resolve_what(what: List[str], client: Client):
```

Comment:
I think, some link could be added here to the format docs, just in case somebody has to adjust this.